### ISSUE TITLE
Adding user defined initialization params

### ISSUE THREAD

**Issue by pierceray** [rwaldron/johnny-five#379]
I would like to add a user defined param into the options of a new Led instance, but it doesn't seem to be added to the options when I try.

Something to the effect of:

```
var led25 = new five.Led({
    pin:13,
    threshold: 25
});
``` 

I'm injecting a new method to the constructor to set the threshold of input value to light up an Led.

```
five.Led.prototype.greaterThanSize = function(value){
    if (value > this.threshold) {
      this.on();
    } else {
      this.off();
    }
  };
```

Is there a better way to achieve this other than the following:
 ```
var led25 = new five.Led(13);
led25["threshold"] = 25;
```

**Comment by dtex** [comment]
If we refactored every class in Johnny-Five to work with:
```
Object.defineProperties(this, opts);
```
... then this would be solved. If that's not realistic then I think we should consider that this needs to be handled in user code, not J5.

**Comment by rwaldron** [comment]
> allow arbitrary user properties via the constructor options, while protecting the properties that the class has known semantics attached to

I still stand by this concern.

**Comment by pierceray** [comment]
It sounds like there are valid concerns for doing it, and it involves a lot of refactoring of the entire library to make it work as well.  In addition no one else has really expressed an interest in adding this kind of functionality.  I'm pretty sure that all of those combined leads us to a "Won't Fix" conclusion.

I'm cool with that, it was a particular case for what I was working on at the time and I thought it might have some utility for others.  Although the "arbitrary code" is still a matter of perspective in my opinion, any user code is arbitrary to some extent.

I'm going to close this issue, as I don't seeing it realistically being a part of the library for the reasons listed previously.

**Comment by scottgonzalez** [comment]
This was just implemented in 8f52736b7d2d0c59b03e3054bde375bf824f1e1c.
